### ISSUE TITLE
CUDA __malloc__ errors on Fedora 35 and Ubuntu Impish

### ISSUE THREAD

**Issue by totaam** [Xpra-org/xpra#3293]
```
CUDA compiling BGRA_to_YUV444   (no file)
 '/opt/cuda-11.4/bin/nvcc' '-fatbin' '-c' 'fs/share/xpra/cuda/BGRA_to_YUV444.cu' '-o' 'fs/share/xpra/cuda/BGRA_to_YUV444.fatbin' '-std=c++03' '-gencode=arch=compute_52,code=sm_52' '-gencode=arch=compute_53,code=sm_53' '-gencode=arch=compute_60,code=sm_60' '-gencode=arch=compute_61,code=sm_61' '-gencode=arch=compute_62,code=sm_62' '-gencode=arch=compute_70,code=sm_70' '-gencode=arch=compute_75,code=sm_75' '-gencode=arch=compute_80,code=sm_80' '-gencode=arch=compute_86,code=sm_86'
Error: failed to compile CUDA kernel BGRA_to_YUV444
 using command: ['/opt/cuda-11.4/bin/nvcc', '-fatbin', '-c', 'fs/share/xpra/cuda/ARGB_to_NV12.cu', '-o', 'fs/share/xpra/cuda/ARGB_to_NV12.fatbin', '-std=c++03', '-gencode=arch=compute_52,code=sm_52', '-gencode=arch=compute_53,code=sm_53', '-gencode=arch=compute_60,code=sm_60', '-gencode=arch=compute_61,code=sm_61', '-gencode=arch=compute_62,code=sm_62', '-gencode=arch=compute_70,code=sm_70', '-gencode=arch=compute_75,code=sm_75', '-gencode=arch=compute_80,code=sm_80', '-gencode=arch=compute_86,code=sm_86']
/usr/include/stdio.h(183): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(195): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(217): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(254): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(279): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(288): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(297): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(303): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(309): error: attribute "__malloc__" does not take arguments
/usr/include/stdio.h(824): error: attribute "__malloc__" does not take arguments
/usr/include/stdlib.h(565): error: attribute "__malloc__" does not take arguments
/usr/include/stdlib.h(569): error: attribute "__malloc__" does not take arguments
/usr/include/stdlib.h(797): error: attribute "__malloc__" does not take arguments
13 errors detected in the compilation of "fs/share/xpra/cuda/ARGB_to_NV12.cu".
```
So CUDA can't deal with the new glibc headers.

**Comment by totaam** [comment]
We skip nvenc on Fedora 35: 059638b4d8581101f1fb261a75ff58d4762a151a.

**Comment by totaam** [comment]
The fix is included in CUDA 11.5.1 and so the commit above re-enables `CUDA` (`nvenc`, `nvfbc` and `nvjpeg` support) for Fedora 35 and Ubuntu Impish.

**Comment by totaam** [comment]
CUDA 11.5.2 works with Fedora 36's GCC 12.
